osu.Game/Overlays/BeatmapListing/Panels/PlayButton.cs

```diff
@@ -39,88 +46,50 @@ public BeatmapSetInfo BeatmapSet
+                    updateEnabledState();
 
-        private Color4 hoverColour;
-        private readonly SpriteIcon icon;
-        private readonly LoadingSpinner loadingSpinner;
-
-        private const float transition_duration = 500;
-
-        private bool loading
-        {
-            set
-            {
-                if (value)
-                {
-                    icon.FadeTo(0.5f, transition_duration, Easing.OutQuint);
-                    loadingSpinner.Show();
-                }
-                else
-                {
-                    icon.FadeTo(1, transition_duration, Easing.OutQuint);
-                    loadingSpinner.Hide();
-                }
+                if (playing.Value)
+                    playing.Value = false;
             }
         }
 
+        [Resolved]
+        private PreviewTrackManager previewTrackManager { get; set; }
+
         public PlayButton(BeatmapSetInfo setInfo = null)
         {
             BeatmapSet = setInfo;
-            AddRange(new Drawable[]
+
+            InternalChild = button = new Button(this)
             {
-                icon = new SpriteIcon
-                {
-                    Anchor = Anchor.Centre,
-                    Origin = Anchor.Centre,
-                    FillMode = FillMode.Fit,
-                    RelativeSizeAxes = Axes.Both,
-                    Icon = FontAwesome.Solid.Play,
-                },
-                loadingSpinner = new LoadingSpinner
-                {
-                    Size = new Vector2(15),
-                },
-            });
+                RelativeSizeAxes = Axes.Both,
+                Action = TogglePlaying,
+            };
 
             playing.ValueChanged += playingStateChanged;
         }
 
-        [Resolved]
-        private PreviewTrackManager previewTrackManager { get; set; }
-
-        [BackgroundDependencyLoader]
-        private void load(OsuColour colour)
+        protected override void LoadComplete()
         {
-            hoverColour = colour.Yellow;
+            base.LoadComplete();
+            Enabled.BindValueChanged(_ => updateEnabledState(), true);
         }
 
-        protected override bool OnClick(ClickEvent e)
+        private void updateEnabledState()
         {
-            playing.Toggle();
-            return true;
-        }
+            var disabled = BeatmapSet == null || !Enabled.Value;
 
-        protected override bool OnHover(HoverEvent e)
-        {
-            icon.FadeColour(hoverColour, 120, Easing.InOutQuint);
-            return base.OnHover(e);
-        }
+            if (playing.Value && disabled)
+                playing.Value = false;
 
-        protected override void OnHoverLost(HoverLostEvent e)
-        {
-            if (!playing.Value)
-                icon.FadeColour(Color4.White, 120, Easing.InOutQuint);
-            base.OnHoverLost(e);
+            playing.Disabled = disabled;
         }
 
+        public void TogglePlaying() => playing.Toggle();
```

Comment:
what's the point? `Playing` (the public one) is... well, public.

Reply:
that would make `Playing.Disabled` mutable, which is not wanted